boost::filesystem replacement, step 1

tomaz82 opened the pull request:
Replace some boost functionality with Utils::FileSystem, this is just step 1 of many to eliminate boost::filesystem

jrassa commented:
This is failing to build for me under Ubuntu.

```
[  4%] Building CXX object es-core/CMakeFiles/es-core.dir/src/Settings.cpp.o
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp: In instantiation of ‘void saveMap(pugi::xml_document&, std::map<K, V>&, const char*) [with K = std::__cxx11::basic_string<char>; V = bool]’:
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:170:50:   required from here
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:154:15: error: no matching function for call to ‘find(std::vector<const char*>::const_iterator, std::vector<const char*>::const_iterator, const std::__cxx11::basic_string<char>&)’
   if(std::find(settings_dont_save.cbegin(), settings_dont_save.cend(), iter->first) != settings_dont_save.cend())
      ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6/bits/locale_facets.h:48:0,
                 from /usr/include/c++/6/bits/basic_ios.h:37,
                 from /usr/include/c++/6/ios:44,
                 from /usr/include/c++/6/istream:38,
                 from /usr/include/c++/6/sstream:38,
                 from /home/jdrassa/Projects/EmulationStation/es-core/src/Log.h:5,
                 from /home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:4:
/usr/include/c++/6/bits/streambuf_iterator.h:369:5: note: candidate: template<class _CharT2> typename __gnu_cxx::__enable_if<std::__is_char<_CharT2>::__value, std::istreambuf_iterator<_CharT> >::__type std::find(std::istreambuf_iterator<_CharT>, std::istreambuf_iterator<_CharT>, const _CharT2&)
     find(istreambuf_iterator<_CharT> __first,
     ^~~~
/usr/include/c++/6/bits/streambuf_iterator.h:369:5: note:   template argument deduction/substitution failed:
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:154:15: note:   ‘__gnu_cxx::__normal_iterator<const char* const*, std::vector<const char*> >’ is not derived from ‘std::istreambuf_iterator<_CharT>’
   if(std::find(settings_dont_save.cbegin(), settings_dont_save.cend(), iter->first) != settings_dont_save.cend())
      ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp: In instantiation of ‘void saveMap(pugi::xml_document&, std::map<K, V>&, const char*) [with K = std::__cxx11::basic_string<char>; V = int]’:
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:171:47:   required from here
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:154:15: error: no matching function for call to ‘find(std::vector<const char*>::const_iterator, std::vector<const char*>::const_iterator, const std::__cxx11::basic_string<char>&)’
In file included from /usr/include/c++/6/bits/locale_facets.h:48:0,
                 from /usr/include/c++/6/bits/basic_ios.h:37,
                 from /usr/include/c++/6/ios:44,
                 from /usr/include/c++/6/istream:38,
                 from /usr/include/c++/6/sstream:38,
                 from /home/jdrassa/Projects/EmulationStation/es-core/src/Log.h:5,
                 from /home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:4:
/usr/include/c++/6/bits/streambuf_iterator.h:369:5: note: candidate: template<class _CharT2> typename __gnu_cxx::__enable_if<std::__is_char<_CharT2>::__value, std::istreambuf_iterator<_CharT> >::__type std::find(std::istreambuf_iterator<_CharT>, std::istreambuf_iterator<_CharT>, const _CharT2&)
     find(istreambuf_iterator<_CharT> __first,
     ^~~~
/usr/include/c++/6/bits/streambuf_iterator.h:369:5: note:   template argument deduction/substitution failed:
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:154:15: note:   ‘__gnu_cxx::__normal_iterator<const char* const*, std::vector<const char*> >’ is not derived from ‘std::istreambuf_iterator<_CharT>’
   if(std::find(settings_dont_save.cbegin(), settings_dont_save.cend(), iter->first) != settings_dont_save.cend())
      ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp: In instantiation of ‘void saveMap(pugi::xml_document&, std::map<K, V>&, const char*) [with K = std::__cxx11::basic_string<char>; V = float]’:
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:172:53:   required from here
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:154:15: error: no matching function for call to ‘find(std::vector<const char*>::const_iterator, std::vector<const char*>::const_iterator, const std::__cxx11::basic_string<char>&)’
In file included from /usr/include/c++/6/bits/locale_facets.h:48:0,
                 from /usr/include/c++/6/bits/basic_ios.h:37,
                 from /usr/include/c++/6/ios:44,
                 from /usr/include/c++/6/istream:38,
                 from /usr/include/c++/6/sstream:38,
                 from /home/jdrassa/Projects/EmulationStation/es-core/src/Log.h:5,
                 from /home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:4:
/usr/include/c++/6/bits/streambuf_iterator.h:369:5: note: candidate: template<class _CharT2> typename __gnu_cxx::__enable_if<std::__is_char<_CharT2>::__value, std::istreambuf_iterator<_CharT> >::__type std::find(std::istreambuf_iterator<_CharT>, std::istreambuf_iterator<_CharT>, const _CharT2&)
     find(istreambuf_iterator<_CharT> __first,
     ^~~~
/usr/include/c++/6/bits/streambuf_iterator.h:369:5: note:   template argument deduction/substitution failed:
/home/jdrassa/Projects/EmulationStation/es-core/src/Settings.cpp:154:15: note:   ‘__gnu_cxx::__normal_iterator<const char* const*, std::vector<const char*> >’ is not derived from ‘std::istreambuf_iterator<_CharT>’
   if(std::find(settings_dont_save.cbegin(), settings_dont_save.cend(), iter->first) != settings_dont_save.cend())
      ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
es-core/CMakeFiles/es-core.dir/build.make:374: recipe for target 'es-core/CMakeFiles/es-core.dir/src/Settings.cpp.o' failed
make[2]: *** [es-core/CMakeFiles/es-core.dir/src/Settings.cpp.o] Error 1
CMakeFiles/Makefile2:217: recipe for target 'es-core/CMakeFiles/es-core.dir/all' failed
make[1]: *** [es-core/CMakeFiles/es-core.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2

```

tomaz82 commented:
My misstake,
I fixed it on my sub-branch on the Pi but forgot to include it into the main PR branch.
Should be fine now, I also squashed the commits.

jrassa commented:
Thanks
